Sources/XCBuildSupport/XcodeBuildSystem.swift

```diff
@@ -149,6 +149,7 @@ public final class XcodeBuildSystem: SPMBuildCore.BuildSystem {
         var settings: [String: String] = [:]
         // Always specify the path of the effective Swift compiler, which was determined in the same way as for the native build system.
         settings["SWIFT_EXEC"] = buildParameters.toolchain.swiftCompiler.pathString
+        settings["LIBRARY_SEARCH_PATHS"] = "$(inherited) \(buildParameters.toolchain.toolchainLibDir.pathString)"
```

Comment:
This is very close! Adding `/swift/macosx` to the string, so we have something that the linker can correctly use, addresses the link failure for me.

```suggestion
        settings["LIBRARY_SEARCH_PATHS"] = "$(inherited) \(buildParameters.toolchain.toolchainLibDir.pathString)/swift/macosx"
```